qa/setup_packages.py

```diff
@@ -60,49 +70,49 @@ def print_install(variant, use, cuda):
     for key in packages.keys():
         if key not in use:
             continue
-        tmp = variant % len(packages[key])
-        val = packages[key][tmp]
+        tmp = variant % len(get_package(packages, key, cuda))
+        val = get_package(packages, key, cuda)[tmp]
         if val == None:
             ret.append(key)
         elif 'http' in val:
             ret.append(get_pyvers_name(val, cuda))
         else:
             ret.append(key + "==" + val)
-        variant = variant // len(packages[key])
+        variant = variant // len(get_package(packages, key, cuda))
     # add all remaining used packages with default versions
     additional = [v for v in use if v not in packages.keys()]
     return " ".join(ret + additional)
 
-def print_remove(use):
+def print_remove(use, cuda):
```

Comment:
nit: I would change this function name to `get_remove_string` or `format_remove`

Reply:
Done